packages/components/multiselect/src/SelectAllOption.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { MultiselectOption } from './MultiselectOption';
+import { getMultiselectStyles } from './Multiselect.styles';
+
+type SelectAllOptionProps = {
+  childNodes: React.ReactNode;
+  iterateOverChildren: (
+    children: React.ReactNode,
+    filter: (child: React.ReactElement) => boolean,
+    callback: (child: React.ReactElement) => void,
+  ) => void;
+};
+
+const callChildEventHandler = (
+  child: React.ReactElement,
+  event: React.ChangeEvent<HTMLInputElement>,
+  checked: boolean,
+) => {
+  if (child.props?.isChecked !== checked && !child.props?.isDisabled) {
+    event.target.value = child.props?.value;
+    event.target.checked = checked;
+    child.props?.onSelectItem(event);
+  }
+};
+
+const areChildrenAllSelected = (
```

Comment:
Also, this looks like copied code from `Multiselect`. If there is time, I think it would b worth putting this into a util file and just import it twice

Reply:
It is very similar to the "iterateOverChildren" functionality which we do to count them. I think i can find a abstraction so we can put that function into a utils file and probaply also merge it with the enriching function. All of them are recursively checking the child components and then do different things with the found options.

Reply:
True, if the generalization is too complex, we can easily skip this. No need to write super intelligent utilities for the sake of DRY :)

Reply:
the only thing they have in common is that they work recursively over the children. which is neccesary because we don t have power over the child components in this component. which gives us some advantags as well as some problems.
maybe this would be the point to question if it is better to have the children as compound components or if we should take the render of the children in our own hands. but that leads then to the same limitations that the autocomplete has.

i think there is no great answer, but i also dislike the high complexity of the solution.